cmake/Modules/CheckCurlSSL.cmake

```diff
@@ -0,0 +1,33 @@
+cmake_push_check_state()
+
+include(CheckCSourceRuns)
+
+set(CMAKE_REQUIRED_LIBRARIES ${CURL_APPEND_LIBS} ${OPENSSL_SSL_LIBRARY})
+set(CMAKE_REQUIRED_INCLUDES ${CURL_INCLUDE_DIR} ${OPENSSL_INCLUDE_DIR})
+
+# suppress warnings, this is a runtime check
+set(CMAKE_REQUIRED_FLAGS -w)
+
+if (NOT WIN32)
+  list(APPEND CMAKE_REQUIRED_LIBRARIES -ldl -lpthread)
+endif()
+
+check_c_source_runs(
+  "#include <stdlib.h>
+  #include <curl/curl.h>
+  int main() {
+    curl_version_info_data* vers = curl_version_info(CURLVERSION_NOW);
+    if (!(vers->features & CURL_VERSION_SSL)) {
+      exit(1);
+    }
+    exit(0);
+  }"
+  TILEDB_CURL_HAS_SSL
+)
+
+if (NOT TILEDB_CURL_HAS_SSL)
+  message(FATAL_ERROR "Target libcurl does not provide SSL -- HTTPS will be broken! "
```

Comment:
Does setting TILEDB_CURL_HAS_SSL work? I thought we need to have that as an option? Also does the failure of the above `check_c_source_runs` override the value of `TILEDB_CURL_HAS_SSL`? On failure `check_c_source_runs` the value of `TILEDB_CURL_HAS_SSL` will be empty, but cmake docs don't specify if it unsets it if it exists or does "empty" just mean that no value is set, so `TILEDB_CURL_HAS_SSL` will remain its previous setting?

Reply:
> so TILEDB_CURL_HAS_SSL will remain its previous setting
My recollection is yes, but deferring this to `1.8.*`